### PR TITLE
Add methods to get manga and chapter url

### PR DESCRIPTION
**Pull request by AntsyLich**
Some fork supports directly opening chapters with webview hence `getChapterUrl`

### CODE REVIEW

File: source-api/src/main/java/eu/kanade/tachiyomi/source/online/HttpSource.kt [hunk 1]
```diff
@@ -356,6 +356,26 @@ abstract class HttpSource : CatalogueSource {
...
     }
 
+    /**
+     * Returns the url of the provided manga
+     *
+     * @param manga the manga
+     * @return url of the manga
+     */
+    open fun getMangaUrl(manga: SManga): String {
+        return "$baseUrl${manga.url}"
```
Comment (alessandrojean) [inline source-api/src/main/java/eu/kanade/tachiyomi/source/online/HttpSource.kt:366]: Maybe make it return the `mangaDetailsRequest(manga).url.toString()` instead?

So for existing sources (that doesn't rely on an API) and have changed it, the new method will already use the correct URL without needing a new update to the extension.

File: source-api/src/main/java/eu/kanade/tachiyomi/source/online/HttpSource.kt [hunk 2]
```diff
@@ -356,6 +356,26 @@ abstract class HttpSource : CatalogueSource {
...
+    }
+
+    /**
+     * Returns the url of the provided chapter
+     *
+     * @param chapter the chapter
+     * @return url of the chapter
+     */
+    open fun getChapterUrl(chapter: SChapter): String {
+        return "$baseUrl${chapter.url}"
```
Comment (alessandrojean) [inline source-api/src/main/java/eu/kanade/tachiyomi/source/online/HttpSource.kt:376]: Probably same here?

File: source-api/src/main/java/eu/kanade/tachiyomi/source/online/HttpSource.kt [hunk 3]
```diff
@@ -356,6 +356,26 @@ abstract class HttpSource : CatalogueSource {
         }
     }
 
+    /**
```
Comment (arkon) [inline source-api/src/main/java/eu/kanade/tachiyomi/source/online/HttpSource.kt:359]: Can you add `@since extensions-lib 1.4` to these?

Comment (AntsyLich) [reply source-api/src/main/java/eu/kanade/tachiyomi/source/online/HttpSource.kt:359]: Done

**Pull request merged**